src/config.ts

```diff
@@ -51,53 +52,83 @@ export default class ConfigService {
    */
   #currentVersion: string | null = null;
 
-  constructor(path: string) {
-    this.#uri = path;
+  constructor(uri: string) {
+    this.#uri = uri;
   }
 
-  async resolve() {
-    let url: URL | null = null;
-
-    try {
-      url = new URL(this.#uri);
-    } catch {
-      // Not a valid URL
+  /**
+   * Fetch and return the configuration module.
+   *
+   * @param deadline When a deadline is provided, fetching the module via HTTP
+   *                 will be aborted if the deadline is exceeded. Specified in
+   *                 milliseconds.
+   */
+  async fetchConfig(deadline: number) {
+    if (!this.configIsViaHttp) {
+      return require(/* webpackIgnore: true */ this.#uri).default;
```

Comment:
This is _very_ confusing to me if I were to use this API. Why not separate this into `getConfig` (or `loadConfig`) and `fetchConfig` and make them one thing, and one thing only.